js/ad_manager_vast.js

```diff
@@ -1101,25 +1372,39 @@ OO.Ads.manager(function(_, $) {
       if (!this.amc || !this.amc.ui || !ad) {
         return;
       }
-      if(params && params.code === this.amc.AD_CANCEL_CODE.TIMEOUT) {
-        failedAd();
-      } else {
-
-        if (params && params.code === this.amc.AD_CANCEL_CODE.SKIPPED && currentAd) {
+      if (params) {
+        if (params.code === this.amc.AD_CANCEL_CODE.TIMEOUT) {
+          failedAd();
+        }
+        else if (params.code === this.amc.AD_CANCEL_CODE.SKIPPED && currentAd) {
           if (currentAd.vpaidAd) {
             // Notify Ad Unit that we are skipping the ad
             _safeFunctionCall(currentAd.vpaidAd, "skipAd");
```

Comment:
Nice job integrating the _skipAd function. If the ad is a vpaid ad, will we be able to handle the skip tracking urls?